plugin/dagger/src/main/java/io/deephaven/plugin/PluginRegistrations.java

```diff
@@ -0,0 +1,9 @@
+package io.deephaven.plugin;
+
+// Provides a concrete class for extra typing information that can be logged during startup
+final class PluginRegistrations extends Registrations {
```

Comment:
I don't understand from the comment alone - why does this class even exist?

Aren't all Registrations about plugins?

Reply:
It's essentially just giving the user more information during startup that the Registration was adapted from the provided Plugin - the user didn't provide the Registration themselves:

```java
    @Provides
    @IntoSet
    static Registration adaptsPlugins(Set<Plugin> plugins) {
        return new PluginRegistrations(plugins);
    }
}
```

This could functionally be the same as:

```java
    @Provides
    @IntoSet
    static Registration adaptsPlugins(Set<Plugin> plugins) {
        return new Registrations(plugins);
    }
}
```
, but then you are "losing" extra type information that `RegistrationModule` can provide about the sourcing of this Registration.

Reply:
Sorry, but why - what is the extra information? I get that the toString itself will be different (I think I saw that it was logged on startup), but wouldnt it equally make sense to have a string arg? or make this an inner class in that module, so that the "this came from _here_" is clear, if that's the point? And what is the user going to do with that information - if the code was left unchanged but with PluginRegistrations renamed to DaggerBindingRegistrations, that might help slightly?

Given that there is a set of `Registrations` (the `s` is missing in your code), and those themselves hold iterables of `Plugin`s, we still don't have traceability of the plugins themselves, is that a concern for someone tracing it?

That is, should `providesServiceLoaderPlugins` itself do some re-wrapping of plugins, so that they can be traced? Otherwise, unless I misunderstand, all `Plugin` instances given to dagger in any way end up in just that one `PluginRegistrations` instance.

Reply:
I see your concerns. Yeah, the extra info was just the class name itself. Re-thinking, adapting each Plugin individually is probably best:

```
server_1      | 2022-01-26T23:02:26.418Z | main                 |  INFO | .d.s.p.PluginRegistration | Registering plugins...
server_1      | 2022-01-26T23:02:26.419Z | main                 |  INFO | .d.s.p.PluginRegistration | Invoking registration: PluginAdapter(Figure:io.deephaven.plot.FigureWidget)
server_1      | 2022-01-26T23:02:26.419Z | main                 |  INFO | .d.s.p.PluginRegistration | Registering object type: Figure / Figure:io.deephaven.plot.FigureWidget
server_1      | 2022-01-26T23:02:26.419Z | main                 |  INFO | .d.s.p.PluginRegistration | Invoking registration: io.deephaven.server.plugin.python.PythonPluginsRegistration@4b21844c
server_1      | 2022-01-26T23:02:27.115Z | main                 |  INFO | .d.s.p.PluginRegistration | Registering object type: matplotlib.figure.Figure / <deephaven.plugin.matplotlib.figure_type.FigureType object at 0x7f019e71cc50>
server_1      | 2022-01-26T23:02:27.116Z | main                 |  INFO | .d.s.p.PluginRegistration | Registering object type: deephaven.plugin.json.Node / <deephaven.plugin.json.JsonType object at 0x7f01609a0400>
server_1      | 2022-01-26T23:02:27.116Z | main                 |  INFO | .d.s.p.PluginRegistration | Registered plugins: objectType=3
```